tests/integration/pilot/ingress/ingress_test.go

```diff
@@ -70,6 +77,105 @@ func TestMain(m *testing.M) {
 		Run()
 }
 
+func TestGateway(t *testing.T) {
+	framework.
+		NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			ns := namespace.NewOrFail(t, ctx, namespace.Config{
+				Prefix: "gateway",
+				Inject: true,
+			})
+			var instance echo.Instance
+			echoboot.NewBuilderOrFail(t, ctx).
+				With(&instance, echo.Config{
+					Service:   "server",
+					Namespace: ns,
+					Subsets:   []echo.SubsetConfig{{}},
+					Pilot:     p,
+					Galley:    g,
+					Ports: []echo.Port{
+						{
+							Name:     "http",
+							Protocol: protocol.HTTP,
+							// We use a port > 1024 to not require root
+							InstancePort: 8090,
+						},
+					},
+				}).
+				BuildOrFail(t)
+			instance.Address()
+			if err := g.ApplyConfig(ns, `
+apiVersion: networking.x.k8s.io/v1alpha1
+kind: GatewayClass
+metadata:
+  name: istio
+spec:
+  controller: istio.io/gateway-controller
+---
+apiVersion: networking.x.k8s.io/v1alpha1
+kind: Gateway
+metadata:
+  name: gateway
+spec:
+  class: istio
+  listeners:
+  - name: primary
+    address:
+      type: NamedAddress
+      value: my.domain.example
+    port: 80
+    protocol: http
+  routes:
+  - group: networking.x-k8s.io/v1alpha1
+    resource: HTTPRoute
+    name: http
+---
+apiVersion: networking.x.k8s.io/v1alpha1
+kind: HTTPRoute
+metadata:
+  name: http
+spec:
+  hosts:
+  - hostname: "my.domain.example"
+    rules:
+    - match:
+        pathType: Prefix
+        path: /get
+      filter:
+        headers:
+          add:
+            my-added-header: added-value
+          remove: [my-removed-header]
```

Comment:
nit: i donot think this is tested

Reply:
+1

Reply:
I'll remove this. I think (at least for now) it makes sense to have this test basic and have the bulk of the detailed coverage as unit tests (under the assumption virtual service integ testing is sufficient, and we can rely on that)